core/integration/storagetest/queue.go

```diff
@@ -0,0 +1,126 @@
+package storagetest
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/keytransparency/core/adminserver"
+	"github.com/google/keytransparency/core/keyserver"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_go_proto"
+)
+
+// Queuer writes items to the queue.
+type Queuer interface {
+	keyserver.MutationLogs
+	adminserver.LogsAdmin
+}
```

Comment:
I'm finding this quite hard to grok. This would seem simpler to me if we split this test into two:
- one which tests the MutationLogs contracts hold up (SetWritable vs ListLogs)
- one which tests the LogsAdmin contracts hold up (Send vs ReadLog)

As it is, this has two test cases which test non-intersecting sets of methods on different interfaces, and introduces the name `Queuer` which it tests, which doesn't appear in any of the code actually under test.

Reply:
Good call. Separated admin and storage test suites